multisig-wallet/project/contracts/multisig-contract/tests/functions/info/compute_hash.rs

```diff
@@ -0,0 +1,180 @@
+
+        let nonce = nonce(&deployer.contract).await.value;
+        let threshold = threshold(&deployer.contract).await.value;
+
+        let threshold_instance = Threshold {
+            contract_identifier: deployer.contract.contract_id().try_into().unwrap(),
+            nonce,
+            threshold,
+        };
+
+        let threshold_instance_token = Token::Struct(vec![
+            threshold_instance.contract_identifier.into_token(),
+            threshold_instance.nonce.into_token(),
+            threshold_instance.threshold.into_token(),
+        ]);
+
+        let encoded_tx_struct = ABIEncoder::encode(&[threshold_instance_token])
+            .unwrap()
+            .resolve(0);
+        let expected_hash = Hasher::hash(encoded_tx_struct);
+
+        let response = compute_hash(
+            &deployer.contract,
+            TypeToHash::Threshold(threshold_instance),
+        )
+        .await
+        .value;
+
+        assert_eq!(Bits256(expected_hash.into()), response);
+    }
+
+    #[tokio::test]
+    async fn gets_transfer_hash() {
+        let (_private_key, deployer, _non_owner) = setup_env(VALID_SIGNER_PK).await.unwrap();
+
+        let nonce = nonce(&deployer.contract).await.value;
+        let target = Identity::Address(deployer.wallet.address().try_into().unwrap());
+        let transfer_params = TransferParams {
+            asset_id: base_asset_contract_id(),
+            value: Some(DEFAULT_TRANSFER_AMOUNT),
+        };
+
+        let transaction_instance = Transaction {
+            contract_call_params: None,
+            contract_identifier: deployer.contract.contract_id().try_into().unwrap(),
+            nonce,
+            target,
+            transfer_params,
+        };
+
+        let transaction_instance_token = Token::Struct(vec![
+            transaction_instance
+                .contract_call_params
+                .clone()
+                .into_token(),
+            transaction_instance.contract_identifier.into_token(),
+            transaction_instance.nonce.into_token(),
+            transaction_instance.target.clone().into_token(),
+            transaction_instance.transfer_params.clone().into_token(),
+        ]);
+
+        let encoded_tx_struct = ABIEncoder::encode(&[transaction_instance_token])
+            .unwrap()
+            .resolve(0);
+        let expected_hash = Hasher::hash(encoded_tx_struct);
+
+        dbg!(Bits256(expected_hash.into()));
+
+        let response = compute_hash(
+            &deployer.contract,
+            TypeToHash::Transaction(transaction_instance),
+        )
+        .await
+        .value;
+
+        assert_eq!(Bits256(expected_hash.into()), response);
+    }
+
+    #[ignore = "Encoding of nested structs with Bytes fields in the RustSDK doesn't match the encoding in Sway. This is not an issue for the use of compute_hash when making a contract call; as the user would not be creating the hash in the SDK"]
```

Comment:
Wasn't the param commented out and therefore this test should run?

Reply:
Your comment below is correct